lib/puppet/type/file/content.rb

```diff
@@ -217,6 +217,22 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
+      uri = URI(source)
+      use_ssl = uri.scheme == 'https'
+      connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
+      connection.request_get(uri.path, add_accept_encoding({"Accept" => "raw"}), &block)
```

Comment:
Sorry for replying extra late @cprice404.

See [above](https://github.com/puppetlabs/puppet/pull/3372#issuecomment-65947429). This is something the agent will do (`apply` does not) in order to avoid slurping the whole file content into a variable prior to writing the whole thing to disk.